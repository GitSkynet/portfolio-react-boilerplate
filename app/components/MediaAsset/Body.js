import styled from 'styled-components';

const Body = styled.div`
  padding: 1vw;
`;

export default Body;
