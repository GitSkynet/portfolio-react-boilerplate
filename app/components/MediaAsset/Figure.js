import PropTypes from 'prop-types';
import styled from 'styled-components';

const Figure = styled.figure`
  align-items: center;
  display: flex;
  justify-content: center;
  order: ${(props) => props.isOdd ? 1 : 2};
  position: relative;
  width: 100%;
  background-color: ${(props) => props.isOdd ? '#0062a3' : '#a36200'};
  border-radius: 0.5rem;

  &::before {
    height: 0;
    left: 50%;
    position: absolute;
    top: 100%;
    transform: translateX(-50%);
    width: 0;
    content: "";
    border-bottom: 0;
    border-left: 0.75em solid transparent;
    border-top: 0.75em solid ${(props) => props.isOdd ? '#0062a3' : '#a36200'};
    border-right: 0.75em solid transparent;

    @media (min-width: 600px) {
      left: ${(props) => props.isOdd ? '100%' : 'inherit'};
      right: ${(props) => props.isOdd ? 'inherit' : '100%'};
      top: inherit;
      transform: translateX(0);
      border-bottom: 0.75em solid transparent;
      border-left: ${(props) => props.isOdd ? '0.75em solid #0062a3' : 0};
      border-top: 0.75em solid transparent;
      border-right: ${(props) => props.isOdd ? 0 : '0.75em solid #a36200'};
    }
  }

  @media (min-width: 600px) {
    width: auto;
  }
`;

Figure.propTypes = {
  isOdd: PropTypes.bool,
};

export default Figure;
