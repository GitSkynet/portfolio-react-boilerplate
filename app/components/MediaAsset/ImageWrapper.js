import ResponsiveImage from '../ResponsiveImage';

const ImageWrapper = ResponsiveImage.extend`
  width: 13.75rem;
  border-radius: .5rem;

  > img {
    border-radius: .5rem;
  }
`;

export default ImageWrapper;
