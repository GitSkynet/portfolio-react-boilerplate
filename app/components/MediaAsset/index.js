import React from 'react';
import PropTypes from 'prop-types';

import Figure from './Figure';
import Body from './Body';
import ImageWrapper from './ImageWrapper';

function MediaAsset(props) {
  return (
    <Figure isOdd={props.isOdd}>
      <Body>
        <ImageWrapper imageSource={props.imageSource} imageAlt="Project" ratio="16by9" />
      </Body>
    </Figure>
  );
}

MediaAsset.propTypes = {
  imageSource: PropTypes.string.isRequired,
  isOdd: PropTypes.bool,
};

export default MediaAsset;
