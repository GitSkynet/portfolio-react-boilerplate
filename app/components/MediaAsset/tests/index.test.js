import React from 'react';
import { shallow } from 'enzyme';

import MediaAsset from '../index';

describe('MediaAsset', () => {
  it('should render and match the snapshot', () => {
    const wrapper = shallow(<MediaAsset imageSource={'/test.png'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt the "idOdd" prop', () => {
    const isOdd = false;
    const wrapper = shallow(
      <MediaAsset imageSource={'/test.png'} isOdd={isOdd} />
    );
    expect(wrapper.prop('isOdd')).toBe(isOdd);
  });
});
