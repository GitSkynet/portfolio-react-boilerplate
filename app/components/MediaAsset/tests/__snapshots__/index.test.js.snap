// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaAsset should render and match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Figure__Figure
    isOdd={undefined}
>
    <Body__Body>
        <Styled(ResponsiveImage)
            imageAlt="Project"
            imageSource="/test.png"
            ratio="16by9"
        />
    </Body__Body>
</Figure__Figure>,
  "nodes": Array [
    <Figure__Figure
      isOdd={undefined}
>
      <Body__Body>
            <Styled(ResponsiveImage)
                  imageAlt="Project"
                  imageSource="/test.png"
                  ratio="16by9"
            />
      </Body__Body>
</Figure__Figure>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <MediaAsset
        imageSource="/test.png"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "imageSource": "/test.png",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Figure__Figure
          isOdd={undefined}
>
          <Body__Body>
                    <Styled(ResponsiveImage)
                              imageAlt="Project"
                              imageSource="/test.png"
                              ratio="16by9"
                    />
          </Body__Body>
</Figure__Figure>,
        "_debugID": 2,
        "_renderedOutput": <Figure__Figure
          isOdd={undefined}
>
          <Body__Body>
                    <Styled(ResponsiveImage)
                              imageAlt="Project"
                              imageSource="/test.png"
                              ratio="16by9"
                    />
          </Body__Body>
</Figure__Figure>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MediaAsset
    imageSource="/test.png"
/>,
}
`;
